src/System.Security.Cryptography.Xml/tests/Samples/SigningVerifyingX509Cert.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using Xunit;
+
+namespace System.Security.Cryptography.Xml.Tests
+{
+    public class SigningVerifyingX509Cert
+    {
+        const string ExampleXml = @"<?xml version=""1.0""?>
+<example>
+<test>some text node</test>
+</example>";
+
+        private static void SignXml(XmlDocument doc, AsymmetricAlgorithm key)
+        {
+            var signedXml = new SignedXml(doc)
+            {
+                SigningKey = key
+            };
+
+            var reference = new Reference();
+            reference.Uri = "";
+
+            reference.AddTransform(new XmlDsigEnvelopedSignatureTransform());
+
+            signedXml.AddReference(reference);
+
+            signedXml.ComputeSignature();
+            XmlElement xmlDigitalSignature = signedXml.GetXml();
+            doc.DocumentElement.AppendChild(doc.ImportNode(xmlDigitalSignature, true));
+        }
+
+        private static bool VerifyXml(string signedXmlText, X509Certificate2 certificate)
+        {
+            XmlDocument xmlDoc = new XmlDocument();
+            xmlDoc.PreserveWhitespace = true;
+            xmlDoc.LoadXml(signedXmlText);
+
+            SignedXml signedXml = new SignedXml(xmlDoc);
+            var signatureNode = (XmlElement)xmlDoc.GetElementsByTagName("Signature")[0];
+            signedXml.LoadXml(signatureNode);
+            return signedXml.CheckSignature(certificate, true);
+        }
+
+        [Fact]
+        public void SignedXmlHasCertificateVerifiableSignature()
+        {
+            using (X509Certificate2 x509cert = TestHelpers.GetSampleX509Certificate())
+            {
+                var xmlDoc = new XmlDocument();
+                xmlDoc.PreserveWhitespace = true;
+                xmlDoc.LoadXml(ExampleXml);
+                SignXml(xmlDoc, x509cert.PrivateKey);
```

Comment:
`GetRSAPrivateKey()` (in a using) is preferred to `PrivateKey`.  On .NET Framework there are observable differences when it comes to RSA-SHA256 (.PrivateKey is always CAPI and frequently gets bound to a CSP which doesn't understand RSA-SHA-2).